lib/Conversion/TritonGPUToLLVM/ControlFlowOpToLLVM.cpp

```diff
@@ -92,7 +92,7 @@ struct CallOpConversion : public ConvertOpToLLVMPattern<triton::CallOp> {
       promotedOperands.push_back(base);
     }
 
-    auto opOffsetAttr = caller->getAttrOfType<mlir::IntegerAttr>(
+    auto opOffsetAttr = callOp->getAttrOfType<mlir::IntegerAttr>(
```

Comment:
The test revealed an additional bug here, we were searching for the offset annotation on the calling function instead of the call op itself so the offset was always defaulting to zero.